standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/TestGetPartitionsUsingProjectionAndFilterSpecs.java

```diff
@@ -229,8 +229,12 @@ public void testPartitionProjectionEmptySpec() throws Throwable {
       Assert.assertEquals(origPartition.getLastAccessTime(), retPartition.getLastAccessTime());
       Assert.assertEquals(origPartition.getSd().getLocation(),
           sharedSD.getLocation() + retPartition.getRelativePath());
-      validateMap(origPartition.getParameters(), retPartition.getParameters());
-      validateList(origPartition.getValues(), retPartition.getValues());
```

Comment:
Were these intended changes? Why are we not validating the latest values with original values?

Reply:
yes, cause we set the exclude key in this test case:
https://github.com/apache/hive/blob/c3fa8a3da5e2e22f58f06356cb90232060404287/standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/TestGetPartitionsUsingProjectionAndFilterSpecs.java#L204

Before the fix, we fallback to regular getPartitions as the project fields are empty:
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L4776